src/dstruct.d

```diff
@@ -238,22 +238,27 @@ extern (C++) class StructDeclaration : AggregateDeclaration
     // For those, today TypeInfo_Struct is generated in COMDAT.
     bool requestTypeInfo;
 
-    final extern (D) this(Loc loc, Identifier id)
+    final extern (D) this(Loc loc, Identifier id, bool inObject)
     {
         super(loc, id);
         zeroInit = 0; // assume false until we do semantic processing
         ispod = ISPODfwd;
+
         // For forward references
         type = new TypeStruct(this);
-        if (id == Id.ModuleInfo && !Module.moduleinfo)
-            Module.moduleinfo = this;
+
+        if (inObject)
+        {
+            if (id == Id.ModuleInfo && !Module.moduleinfo)
```

Comment:
Also, in the same PR I linked above, I also add detection for `Object.Interface`.  Which I shall forward upstream as soon as this is in.

Do we really want to deny people from creating their own `struct Interface` just because it appears in object.d?

I say, that - however this fails to compile anyway:
```
import b;
Interface collision;
```
```
struct Interface { int field; }
```
```
error: object.Interface at object.d:184:1 conflicts with b.Interface
```